libsession/src/main/java/org/session/libsession/messaging/sending_receiving/ReceivedMessageHandler.kt

```diff
@@ -148,11 +155,30 @@ fun MessageReceiver.cancelTypingIndicatorsIfNeeded(senderPublicKey: String) {
 }
 
 private fun MessageReceiver.handleExpirationTimerUpdate(message: ExpirationTimerUpdate) {
-    if (message.duration!! > 0) {
-        SSKEnvironment.shared.messageExpirationManager.setExpirationTimer(message)
-    } else {
-        SSKEnvironment.shared.messageExpirationManager.disableExpirationTimer(message)
+    if (ExpirationConfiguration.isNewConfigEnabled) return
+    val module = MessagingModuleConfiguration.shared
+    val recipient = Recipient.from(module.context, Address.fromSerialized(message.sender!!), false)
+    val type = when {
+        recipient.isContactRecipient -> ExpiryMode.AfterRead(message.duration!!.toLong())
+        recipient.isGroupRecipient -> ExpiryMode.AfterSend(message.duration!!.toLong())
+        else -> ExpiryMode.NONE
+    }
+    try {
+        var threadId: Long = module.storage.getOrCreateThreadIdFor(fromSerialized(message.sender!!))
+        if (message.groupPublicKey != null) {
+            threadId = module.storage.getOrCreateThreadIdFor(fromSerialized(doubleEncodeGroupID(message.groupPublicKey!!)))
+        }
+        module.storage.setExpirationConfiguration(
+            ExpirationConfiguration(
+                threadId,
+                type,
+                SnodeAPI.nowWithOffset
```

Comment:
Should we use the message's time here instead of the current time? I'm not 100% familiar with what we are tracking here but might be right